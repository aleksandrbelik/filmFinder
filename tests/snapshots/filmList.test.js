import React from 'react';
import FilmsList from 'Common/filmsList/filmsList';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<FilmsList />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly for one film in the list', () => {
  const tree = renderer
    .create(<FilmsList films={[
      {
        name: 'bla bla la ',
        genre: 'ssdfsdfsdfdsf',
        year: '2004',
        imgSrc: 'https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782'
      }
    ]}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly for five film in the list', () => {
  const tree = renderer
    .create(<FilmsList films={[
      {
        name: 'bla bla la ',
        genre: 'ssdfsdfsdfdsf',
        year: '2004',
        imgSrc: 'https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782'
      },
      {
        name: 'bla bla la 1',
        genre: 'ssdfsdfsdfdsf',
        year: '2004',
        imgSrc: 'https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782'
      },
      {
        name: 'bla bla la 2',
        genre: 'ssdfsdfsdfdsf',
        year: '2004',
        imgSrc: 'https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782'
      },
      {
        name: 'bla bla la 3',
        genre: 'ssdfsdfsdfdsf',
        year: '2004',
        imgSrc: 'https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782'
      },
      {
        name: 'bla bla la 4',
        genre: 'ssdfsdfsdfdsf',
        year: '2004',
        imgSrc: 'https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782'
      }
    ]}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
