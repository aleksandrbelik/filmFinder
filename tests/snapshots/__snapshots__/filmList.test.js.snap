// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="film-list"
>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 1
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 2
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 3
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 4
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 5
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 6
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 7
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 8
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 9
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 10
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 11
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 12
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
</div>
`;

exports[`renders correctly for five film in the list 1`] = `
<div
  className="film-list"
>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 1
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 2
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 3
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 4
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
</div>
`;

exports[`renders correctly for one film in the list 1`] = `
<div
  className="film-list"
>
  <div
    className="film-card"
  >
    <img
      alt=""
      className="film-card__image"
      src="https://boygeniusreport.files.wordpress.com/2017/01/cat.jpg?quality=98&strip=all&w=782"
    />
    <div
      className="film-card__info"
    >
      <div
        className="film-card__name"
      >
        bla bla la 
      </div>
      <div
        className="film-card__year"
      >
        2004
      </div>
    </div>
    <div
      className="film-card__genre"
    >
      ssdfsdfsdfdsf
    </div>
  </div>
</div>
`;
